aten/src/ATen/native/Convolution.cpp

```diff
@@ -422,7 +422,13 @@ struct ConvParams {
   // that maybe the compiledWithCuDNN() check sometimes segfaults (though I can't imagine how)
 #if !defined(C10_MOBILE)
     if (needs_64bit_indexing_no_split(input, weight)) {
-      return false;
+      static long cudnn_version = detail::getCUDAHooks().versionCuDNN();
+      if (!(cudnn_version >= 90300 && at::native::cudnnv8_enabled_check_debug())) {
+        TORCH_WARN_ONCE("cuDNN cannot be used for large non-batch-splittable convolutions"
+                        " if the V8 API is not enabled or before cuDNN version 9.3+."
```

Comment:
Are we sure about this `or`? Or do we need `and`?

Reply:
/cc @Skylion007

Reply:
Are you referring to the warning message? You can apply De Morgan's law to translate the predicate to if cudnn_version < 9.3 _or_ the V8 API is disabled, then non-batch-splittable 64bit cases cannot be dispatched to cuDNN.

Reply:
Yes but I am still entering in the condition. What is then the condition in https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/ConvUtils.h#L80 ?

Reply:
@eqy Isn't the logic wrong currently? See https://github.com/pytorch/pytorch/blob/main/c10/util/env.cpp#L93

Reply:
Check https://github.com/pytorch/pytorch/pull/139963 if it could help